pymodbus/server/reactive/main.py

```diff
@@ -177,6 +183,20 @@ def getValues(self, fc_as_hex, address, count=1):
                     #              "'%s'", _block_type, values, address)
                     self.store[_block_type].setValues(address, values)
                 self._read_counter[_block_type] += 1
+        elif self._change_rate > 0 and _block_type in {"d", "i"}:
+            for offset in range(count):
+                if random.randint(1, 100) <= self._change_rate:
```

Comment:
@hahattan Thanks for this PR, would you mind giving a short description with examples (if possible) how does this work?

Reply:
Hi, 

So I have a use case that an client is reading datas from repl-server and sending them to somewhere else if the value has changed.
I was using the `-r/--random` to simulate this scenario but the result is not what I want:
For example if I'm reading 1000 registers and expecting their values have 20% chance being updated. I used `--random 5`, the expected data sent by client would be `0,0,0,0,1000,0,0,0,0,1000 ....`. (what I want is something like `200,200,200,200,200...`)

Reply:
So if I understand this feature correctly, you are trying to return random values for a percentage of input registers/ discrete inputs being read. If I am reading 10 registers and have set `-c=20` then only 2 out of the 10 registers are expected to change the value, even though I have passed `--random=1` which means randomise values for every read. Is my understanding correct?

Reply:
Not exactly, I made `--random` overwrite `--change-rate` because `--random` IMO is more predictable:
```python
        if self._randomize > 0:
            self._change_rate = 0
```

So if user provide both `-r` and `-c` only `-r` will work

Reply:
Got it, so if `-r` is not passed , then my understanding of how `-c` should works is good?

Reply:
Yes

Reply:
In any case how does `if random.randint(1, 100) <= self._change_rate:` ensure that we are going to change only the required percentage of registers/discrete inputs being changed. `random.randint` is not deterministic and there might be cases where the number of registers being changed are above and below the required count. Is this acceptable?

How about calculating and rounding off the number of registers to be changed and randomly update values?

```
elif self._change_rate > 0 and _block_type in {"d", "i"}:
    changed_indicies = []
    regs_to_changes = round(count * self._change_rate/100)
    random_indices = random.sample(count, regs_to_changes)
    for offset in random_indices:
        with self._lock:
            # Update values
            if _block_type == "d":
                min_val = self._min_binary_value
                max_val = self._max_binary_value
            else:
                min_val = self._min_register_value
                max_val = self._max_register_value
            self.store[_block_type].setValues(
                address + offset, random.randint(min_val, max_val)
```

Reply:
> and there might be cases where the number of registers being changed are above and below the required count. Is this acceptable?

Yes, this is acceptable for me. Sorry for misunderstanding what you meant in previous comment.

> 
> How about calculating and rounding off the number of registers to be changed and randomly update values?
> 
> ```
> elif self._change_rate > 0 and _block_type in {"d", "i"}:
>     changed_indicies = []
>     regs_to_changes = round(count * self._change_rate/100)
>     random_indices = random.sample(count, regs_to_changes)
>     for offset in random_indices:
>         with self._lock:
>             # Update values
>             if _block_type == "d":
>                 min_val = self._min_binary_value
>                 max_val = self._max_binary_value
>             else:
>                 min_val = self._min_register_value
>                 max_val = self._max_register_value
>             self.store[_block_type].setValues(
>                 address + offset, random.randint(min_val, max_val)
> ```

I'm okay with this solution as well, if there's no other comment I can push the commit with suggested changes.

Reply:
Yes, if you are OK please use the suggested changes. Thanks